GitUI/UserControls/FileStatusList.cs

```diff
@@ -173,60 +168,6 @@ public new void Focus()
             }
         }
 
-        public void BeginUpdate()
-        {
-            FileStatusListView.BeginUpdate();
-        }
-
-        public void EndUpdate()
-        {
-            FileStatusListView.EndUpdate();
-        }
-
-        private string GetItemText(Graphics graphics, GitItemStatus gitItemStatus)
-        {
-            var pathFormatter = new PathFormatter(graphics, FileStatusListView.Font);
-
-            return pathFormatter.FormatTextForDrawing(FileStatusListView.ClientSize.Width - ImageSize,
-                                                      gitItemStatus.Name, gitItemStatus.OldName);
-        }
-
-        private void FileStatusListView_DrawItem(object sender, DrawListViewItemEventArgs e)
-        {
-            if (e.Bounds.Height <= 0 || e.Bounds.Width <= 0 || e.ItemIndex < 0)
-                return;
-
-            e.DrawBackground();
-            Color color;
-            if (e.Item.Selected)
-            {
-                e.Graphics.FillRectangle(SystemBrushes.Highlight, e.Bounds);
-                color = SystemColors.HighlightText;
-            }
-            else
-                color = SystemColors.WindowText;
-            e.DrawFocusRectangle();
-
-            e.Graphics.FillRectangle(Brushes.White, e.Bounds.Left, e.Bounds.Top, ImageSize, e.Bounds.Height);
-
-            int centeredImageTop = e.Bounds.Top;
-            if ((e.Bounds.Height - ImageSize) > 1)
-                centeredImageTop = e.Bounds.Top + ((e.Bounds.Height - ImageSize) / 2);
-
-            var image = e.Item.ImageList.Images[e.Item.ImageIndex];
-
-            if (image != null)
-                e.Graphics.DrawImage(image, e.Bounds.Left, centeredImageTop, ImageSize, ImageSize);
-
-            GitItemStatus gitItemStatus = (GitItemStatus)e.Item.Tag;
-
-            string text = GetItemText(e.Graphics, gitItemStatus);
```

Comment:
This is a regression - `GetItemText` makes a call to `PathFormatter` which truncates paths as per user configurations.
The new controls must continue rendering formatted text

![image](https://user-images.githubusercontent.com/4403806/36425366-adae818c-169a-11e8-851e-588620a63dae.png)

Reply:
I guess I was a bit aggressive in removing this owner-draw code (I wrongly assumed that it was only about not drawing the selected colour around the icon). I'll try to get this working nicely, otherwise I'll revert the code for this control, too.

Reply:
I've worked around this by using the existing methods for getting the appropriately truncated text but leaving the actual drawing up to Windows. I've tested this with all 4 truncation options.